### PR TITLE
feature/317_model_entities_parsing

### PR DESCRIPTION
**Pull request by frbattid**
Implements issue https://github.com/telefonicaid/fiware-connectors/issues/317

Really, nothing has been implemented regarding the model estities parsing since this was something already implemented. What has been done was to implement a testing purpose sink, `OrionTestSink` in charge of printing (logging) the content of a parsed notification; if this printing is possible, that means the notification was properly parsed.

In addition, a `OrionTestSink.md` document has been created (in the future, all the sinks must have an associated document as this one).

```
2015-02-13 20:47:29,218 (1503175079@qtp-636090328-0) [INFO - es.tid.fiware.fiwareconnectors.cygnus.handlers.OrionRestHandler.getEvents(OrionRestHandler.java:153)] Starting transaction (1423856842-608-0000000000)
2015-02-13 20:47:29,220 (1503175079@qtp-636090328-0) [INFO - es.tid.fiware.fiwareconnectors.cygnus.handlers.OrionRestHandler.getEvents(OrionRestHandler.java:239)] Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "id": "netrisModel",        "isPattern": "false",        "type": "model",        "attributes": [          {            "name": "TimeInstant",            "type": "ISO8601",            "value": "2014-03-20T14:21:14Z"          },          {            "name": "ModelProp1",            "type": "String",            "value": "abc"          },          {            "name": "ModelProp2",            "type": "String",            "value": "def"          },          {            "name": "te",            "type": "compound",            "value":              {                "name": "Temperatura Exterior",                "phenomenon": "temperature",      	        "type": "Quantity",                "uom": "Cel",                "persistence": "sth,ckan"              }          },          {            "name": "ti",            "type": "compound",            "value": [              {                "name": "Temperatura Interior",                "phenomenon": "temperature",                "type": "Quantity",                "uom": "Cel"              }            ]          },          {            "name": "h",            "type": "compound",            "value": [              {                "name": "Humedad casa",                "phenomenon": "humidity",      	        "type": "Quantity",                "uom": "%"              }            ]          },          {            "name": "reset",            "type": "compound",            "value": [              {                "type": "action",                "input_params": "reqTimeInstant,correlatorID,param_a,param_b",                "output_params": "result,resTimeIntant,param_c,param_d"              }            ]          },          {            "name": "color",            "type": "compound",            "value": [              {                "type": "action",                "input_params": "reqTimeInstant,correlatorID",                "output_params": "result,resTimeInstant"              }            ]          }        ]      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
2015-02-13 20:47:29,222 (1503175079@qtp-636090328-0) [INFO - es.tid.fiware.fiwareconnectors.cygnus.handlers.OrionRestHandler.getEvents(OrionRestHandler.java:261)] Event put in the channel (id=863204900, ttl=10)
2015-02-13 20:47:29,285 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionSink.process(OrionSink.java:126)] Event got from the channel (id=863204900, headers={fiware-servicepath=def_servpath, destination=netrismodel_model, content-type=application/json, fiware-service=def_serv, ttl=10, transactionId=1423856842-608-0000000000, timestamp=1423856849222}, bodyLength=2227)
2015-02-13 20:47:29,289 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:77)] [test-sink] Processing headers (recvTimeTs=1423856849222 (2015-02-13T20:47:29.222), fiwareService=def_serv, fiwareServicePath=def_servpath, destinations=[netrismodel_model])
2015-02-13 20:47:29,289 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:90)] [test-sink] Processing context element (id=netrisModel, type= model)
2015-02-13 20:47:29,289 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:109)] [test-sink] Processing context attribute (name=TimeInstant, type=ISO8601, value=2014-03-20T14:21:14Z, metadata=[])
2015-02-13 20:47:29,289 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:109)] [test-sink] Processing context attribute (name=ModelProp1, type=String, value=abc, metadata=[])
2015-02-13 20:47:29,290 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:109)] [test-sink] Processing context attribute (name=ModelProp2, type=String, value=def, metadata=[])
2015-02-13 20:47:29,294 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:109)] [test-sink] Processing context attribute (name=te, type=compound, value={"name":"Temperatura Exterior","phenomenon":"temperature","type":"Quantity","uom":"Cel","persistence":"sth,ckan"}, metadata=[])
2015-02-13 20:47:29,294 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:109)] [test-sink] Processing context attribute (name=ti, type=compound, value=[{"name":"Temperatura Interior","phenomenon":"temperature","type":"Quantity","uom":"Cel"}], metadata=[])
2015-02-13 20:47:29,294 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:109)] [test-sink] Processing context attribute (name=h, type=compound, value=[{"name":"Humedad casa","phenomenon":"humidity","type":"Quantity","uom":"%"}], metadata=[])
2015-02-13 20:47:29,294 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:109)] [test-sink] Processing context attribute (name=reset, type=compound, value=[{"type":"action","input_params":"reqTimeInstant,correlatorID,param_a,param_b","output_params":"result,resTimeIntant,param_c,param_d"}], metadata=[])
2015-02-13 20:47:29,294 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionTestSink.persist(OrionTestSink.java:109)] [test-sink] Processing context attribute (name=color, type=compound, value=[{"type":"action","input_params":"reqTimeInstant,correlatorID","output_params":"result,resTimeInstant"}], metadata=[])
2015-02-13 20:47:29,294 (SinkRunner-PollingRunner-DefaultSinkProcessor) [INFO - es.tid.fiware.fiwareconnectors.cygnus.sinks.OrionSink.process(OrionSink.java:187)] Finishing transaction (1423856842-608-0000000000)
```

### CODE REVIEW

File: flume/doc/design/OrionTestSink.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# OrionTestSink
```
Comment (fgalan) [inline flume/doc/design/OrionTestSink.md]: What about the existing sinks for HDFS, CKAN and MySQL? Have documentation being refactor to include their information as separate .md files as this one?

File: flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionTestSink.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2014 Telefonica Investigación y Desarrollo, S.A.U
```
Comment (fgalan) [inline flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionTestSink.java]: Step year to 2015 in new files.

File: flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionMySQLSink.java [hunk 3]
```diff
@@ -66,7 +66,7 @@
      */
     public OrionMySQLSink() {
         super();
-        logger = CygnusLogger.getLogger(OrionHDFSSink.class);
+        logger = CygnusLogger.getLogger(OrionMySQLSink.class);
```
Comment (fgalan) [inline flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionMySQLSink.java]: This seems to be a bugfix... It has any impact? Or it happened that the logger for HDFS sink and SQL were the same so it is a small bugfix?

File: resources/ngsi-examples/notification-json-modelentity.sh [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            "name": "ModelProp2",
+            "type": "String",
+            "value": "def"
+          },
+          {
+            "name": "te",
+            "type": "compound",
+            "value":
+              {
+                "name": "Temperatura Exterior",
```
Comment (fgalan) [inline resources/ngsi-examples/notification-json-modelentity.sh]: Maybe this should be in English?

File: resources/ngsi-examples/notification-json-modelentity.sh [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                "phenomenon": "temperature",
+      	        "type": "Quantity",
+                "uom": "Cel",
+                "persistence": "sth,ckan"
```
Comment (fgalan) [inline resources/ngsi-examples/notification-json-modelentity.sh]: I think we talked in the past to use a vector for this, i.e.:

```
"persistence": [ "sth", "ckan" ]
```

File: resources/ngsi-examples/notification-json-modelentity.sh [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+              }
+          },
+          {
+            "name": "ti",
+            "type": "compound",
+            "value": [
+              {
+                "name": "Temperatura Interior",
```
Comment (fgalan) [inline resources/ngsi-examples/notification-json-modelentity.sh]: Maybe this should be in English?

File: resources/ngsi-examples/notification-json-modelentity.sh [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+                "uom": "Cel"
+              }
+            ]
+          },
+          {
+            "name": "h",
+            "type": "compound",
+            "value": [
+              {
+                "name": "Humedad casa",
```
Comment (fgalan) [inline resources/ngsi-examples/notification-json-modelentity.sh]: Maybe this should be in English?

File: flume/doc/design/OrionTestSink.md (hunk 1 above)
Comment (frbattid) [inline flume/doc/design/OrionTestSink.md]: Yes, at any time in the future I would like to make such a refactor, or at least, to document them with the same sections (functionallity, detailed configuration, uses cases).

File: flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionMySQLSink.java (hunk 3 above)
Comment (frbattid) [inline flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionMySQLSink.java]: The SQL sink was logging as the HDFS sink... nothing critical. Anyway, I'll made a fix in release/0.6.0 before tagging to release/iotplatform-v1.1

File: resources/ngsi-examples/notification-json-modelentity.sh (hunk 7 above)
Comment (frbattid) [inline resources/ngsi-examples/notification-json-modelentity.sh]: Fixed in https://github.com/telefonicaid/fiware-connectors/commit/fc3ac6b227108efd40148ab1ac465d28d81e896f

File: resources/ngsi-examples/notification-json-modelentity.sh (hunk 6 above)
Comment (frbattid) [inline resources/ngsi-examples/notification-json-modelentity.sh]: Fixed in https://github.com/telefonicaid/fiware-connectors/commit/fc3ac6b227108efd40148ab1ac465d28d81e896f

File: resources/ngsi-examples/notification-json-modelentity.sh (hunk 5 above)
Comment (frbattid) [inline resources/ngsi-examples/notification-json-modelentity.sh]: Fixed in https://github.com/telefonicaid/fiware-connectors/commit/fc3ac6b227108efd40148ab1ac465d28d81e896f

File: resources/ngsi-examples/notification-json-modelentity.sh (hunk 4 above)
Comment (frbattid) [inline resources/ngsi-examples/notification-json-modelentity.sh]: Fixed in https://github.com/telefonicaid/fiware-connectors/commit/fc3ac6b227108efd40148ab1ac465d28d81e896f

File: flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionTestSink.java (hunk 2 above)
Comment (frbattid) [inline flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionTestSink.java]: Fixed in https://github.com/telefonicaid/fiware-connectors/commit/fc3ac6b227108efd40148ab1ac465d28d81e896f

File: flume/doc/design/OrionTestSink.md (hunk 1 above)
Comment (fgalan) [inline flume/doc/design/OrionTestSink.md]: If the refactoring task is pending for the future, I'd suggest to create an issue for that.

File: flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionMySQLSink.java (hunk 3 above)
Comment (fgalan) [inline flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionMySQLSink.java]: If the fix is meanifull, an entry should be added to CHANGES_NEXT_REELASE.

I don't understand what you mean by "tagging to release/ioteplatform-v1.1" but maybe we can dissuss that part out-of-band (by skype).

Comment (frbattid) [inline flume/src/main/java/es/tid/fiware/fiwareconnectors/cygnus/sinks/OrionMySQLSink.java]: Issue create for release/0.6.0 --> https://github.com/telefonicaid/fiware-connectors/issues/321

CHANGES_NEXT_RELEASE updated in https://github.com/telefonicaid/fiware-connectors/commit/758f0c512a5bc3a8674d3f724e9fb5bae22d8ae3

File: flume/doc/design/OrionTestSink.md (hunk 1 above)
Comment (frbattid) [inline flume/doc/design/OrionTestSink.md]: Added --> https://github.com/telefonicaid/fiware-connectors/issues/322

File: flume/CHANGES_NEXT_RELEASE [hunk 8]
```diff
@@ -5,3 +5,5 @@
 * Encourage the usage of FQDNs regarding HDFS namenodes when using Kerberos (#309)
 * Fixed the RoundRoundChannelSelector, now it works for multiple storages (#298)
 * Allow for infinite TTL (#236)
+* Added OrionTestSink (#307)
+* Fixed OrionMySQLSink logger (now it logs as OrionMySQLSink, not as OrionHDFSSink)
```
Comment (fgalan) [inline flume/CHANGES_NEXT_RELEASE]: Given you have created an issue for this, maybe is a good idea to use it in the text, i.e.:

```
Fixed OrionMySQLSink logger (now it logs as OrionMySQLSink, not as OrionHDFSSink) (#321)
```

Comment (frbattid) [inline flume/CHANGES_NEXT_RELEASE]: Fixed in https://github.com/telefonicaid/fiware-connectors/commit/26de0cb95ca537a027191777742cbad2b2e3aae7

**Comment by fgalan** [comment]
LGTM

**Pull request merged**